File: backend/FinanceApp.Domain/Entities/Supplier.cs
namespace FinanceApp.Domain.Entities;

public class Supplier : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? ShortName { get; set; }
    public string? ContactPerson { get; set; }
    public string? ContactPhone { get; set; }
    public string? ContactEmail { get; set; }
    public string? Address { get; set; }
    public string? TaxNumber { get; set; }
    public string? BankAccount { get; set; }
    public string? BankName { get; set; }
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public ICollection<TagBinding> TagBindings { get; set; } = new List<TagBinding>();
}

File: frontend/src/features/master-data/suppliers/types/supplier.ts
export interface Supplier {
  id: number
  name: string
  shortName?: string
  contactPerson?: string
  contactPhone?: string
  contactEmail?: string
  address?: string
  taxNumber?: string
  bankAccount?: string
  bankName?: string
  description?: string
  isActive: boolean
  createdAt: string
}

export interface SupplierStatistics {
  totalCount: number
  activeCount: number
  thisMonthNewCount: number
  inactiveCount: number
}

export interface CreateSupplierRequest {
  name: string
  shortName?: string
  contactPerson?: string
  contactPhone?: string
  contactEmail?: string
  address?: string
  taxNumber?: string
  bankAccount?: string
  bankName?: string
  description?: string
}

export interface UpdateSupplierRequest {
  name: string
  shortName?: string
  contactPerson?: string
  contactPhone?: string
  contactEmail?: string
  address?: string
  taxNumber?: string
  bankAccount?: string
  bankName?: string
  description?: string
  isActive: boolean
}

export interface SupplierFinanceSummary {
  totalReceivable: number
  totalReceived: number
  receivableRemaining: number
  receivableOverdueCount: number
  receivableOverdueAmount: number
  totalPayable: number
  totalPaid: number
  payableRemaining: number
  payableOverdueCount: number
  payableOverdueAmount: number
  projectCount: number
}
